tests/virt_autotest/sriov_pci_passthrough.pm

```diff
@@ -0,0 +1,279 @@
+# SUSE's openQA tests
+ #
+ # Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+ 
+# Summary: A test on KVM host to pass a SR-IOV NIC VF to guest via libvirt.
+# Maintainer: Julie CAO <JCao@suse.com>
+ 
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use set_config_as_glue;
+use utils;
+use virt_utils;
+use xen;
+use virt_autotest::virtual_network_utils qw(save_guest_ip test_network_interface);
+ 
+sub run_test {
+  my $self = shift;
+
+#  my $hypervisor;
+#  if (check_var("HOST_HYPERVISOR", "kvm") || check_var("SYSTEM_ROLE", "kvm") || check_var("HYPERVISOR", "kvm")) {
+#    $hypervisor = "kvm";
+#  }
+#  else {
+#    $hypervisor = "xen";
+#  }
+
+  #get the SR-IOV device BDF and interface 
+  my @host_pfs;
+  @host_pfs = get_sriov_devices ();
+  $self->{test_results}->{"host"}->{"Error: not found any SR-IOV network device in the host!"}->{status} = 'FAILED' if ! $#host_pfs;
+  record_info("Find SR-IOV devices", "@host_pfs");
+  
+  # enable 8 vfs for the SR-IOV device on host
+  my @host_vfs = enable_vf(@host_pfs);
+  $self->{test_results}->{"host"}->{"Error: fail to enable the SR-IOV PF @host_pfs!"}->{status} = 'FAILED' if ! @host_vfs;
+  record_info("VFs enabled", "@host_vfs");
+
+  #generate ssh key on host
+  script_run "rm ~/.ssh/id_rsa*; ssh-keygen -t rsa -f ~/.ssh/id_rsa -P ''";
+
+  foreach my $guest (keys %xen::guests) {
+  my $guest = "sles-15-sp2-64-fv-def-net";
```

Comment:
Just confirm sles-15-sp2-64-fv-def-net guest, right?

Reply:
So far, I plan to support the test since sles15sp2. because the xen 4.13 differentiate 4.12 much. we have to create two sets of tests for older release. Also, Alice tested sles12sp5 manually.

Reply:
Quick idea, there should be confirmed both PV and FV guest together on xen host(this test run should support on xen host here), just only confirm sles-15-sp2-64-fv-def-net guest not enough. 
So, please add some helpful info here as comments

Reply:
XEN will be supported quite soon. working on it now...

Reply:
@Julie-CAO I thought this line was for your debug testing only. Since you share your support plan, maybe I should have more explanation here.
PCI passthru tests are considered as important tests, which antoine gives P2 priority after consulting virtualization product manager's opinion. So it has only lower priority than guest install and guest migration tests, but higher than upgrade tests and all other feature tests. We need to emphasize this test. 
IMO,  for middle term, we should support this automation test for our big host/guest matrix just as prj1 does
 (hotplugging/virtual network etc do similarly). If splitting it into phases, as first step, we should support it for sle15sp2+ host with the big guests list as prj1. Then sle12/sle11 hosts.

Reply:
Oh, thanks, alice. I did not realize this point. In this case, I really need to do more work, including supporting more guests, more verification runs and more complex test scenarios.